test/dataframe.jl

```diff
@@ -707,6 +707,35 @@ module TestDataFrame
         @test names(df) == [:x3, :x3_1, :x3_2, :x4]
     end
 
+    @testset "setindex! sets row with dict/iterable" begin
+        df = DataFrame(a=[0, 0], b=['a', 'a'])
+        dfa = DataFrame(a=0:1, b='a':'b')
+        df[2, :] = Dict(:a => 1, :b => 'b')
+        @test df == dfa
+
+        df = DataFrame(a=[0, 0], b=['a', 'a'])
+        df[2, :] = (a=1, b='b')
```

Comment:
Put `b` first to check that order doesn't matter.
```suggestion
        df[2, :] = (b='b', a=1)
```